server/client.go

```diff
@@ -3216,6 +3260,15 @@ func (c *client) processInboundMsg(msg []byte) {
 	}
 }
 
+// selectMappedSubject will chose the mapped subject based on the client's inbound subject.
+func (c *client) selectMappedSubject() bool {
+	nsubj, changed := c.acc.selectMappedSubject(string(c.pa.subject))
+	if changed {
+		c.pa.subject = []byte(nsubj)
+	}
+	return changed
+}
+
```

Comment:
Note:
For an inbound tee functionality I would support lists of (weighted) mappings.
mappings = { foo: [bar, baz] } -> send all to bar and baz
mappings = { foo: [foo, baz] } -> send all to foo and baz
mappings = { foo: [bar, [{ dest: baz, weight: 1% }]} -> send all to bar and 1% of traffic to baz (I'd see sampling use cases here)

selectMappedSubject would return more than one subject.
when processing MSG_END_N I would just iterate over that list and assign 
for subj, _ := range c.pa.subjects {
c.pa.subject = subj
if trace {
    c.traceMsg(c.msgBuf)
}
c.processInboundMsg(c.msgBuf)
}

Reply:
I think we can reuse a bunch of this for `tee` and `filter` but will leave for now. I would probably make separate config section TBH.

`tee = {
    foo: bar
    baz: [ baz.1, baz.2]
}`

Reply:
FYI checked account reload, if any account section is defined in config we always reloadAuthorization which includes accounts, so no change in behavior with mappings being added.

Reply:
> I think we can reuse a bunch of this for `tee` and `filter` but will leave for now. I would probably make separate config section TBH.
> 
> `tee = { foo: bar baz: [ baz.1, baz.2] }`

That will raise the question when is it applied in relation to mappings and will have to be explained.
If you only have mappings from one subject to one or more other subjects, some of which can induce loss that question won't come up.
I get that a [] is easily missed. How about an object instead? {tee: []} where the list consists of (weighted) mappings. 
mappings = { foo: {tee: [bar, { dest: baz, weight: 1% }}}

Reply:
Possibly, we can punt tee (and filter) to another PR for sure.